library/src/main/java/io/ona/kujaku/services/MapboxOfflineDownloaderService.java

```diff
@@ -151,14 +188,62 @@ private boolean persistOfflineMapTask(@Nullable Intent intent) {
 
                         return true;
                     }
-                } else {
+                } else if (serviceAction == SERVICE_ACTION.DELETE_MAP){
                     MapBoxDeleteTask deleteTask = new MapBoxDeleteTask();
                     deleteTask.setMapBoxAccessToken(mapBoxAccessToken);
                     deleteTask.setMapName(mapUniqueName);
 
                     MapBoxDeleteTask.constructMapBoxOfflineQueueTask(deleteTask);
 
                     return true;
+                } else {
+                    final String taskType = extras.getString(Constants.PARCELABLE_KEY_DELETE_TASK_TYPE, "");
+
+                    if (!TextUtils.isEmpty(taskType)) {
+                        if (taskType.equals(MapBoxOfflineQueueTask.TASK_TYPE_DOWNLOAD)) {
+                            MapBoxOfflineResourcesDownloader mapBoxOfflineResourcesDownloader = MapBoxOfflineResourcesDownloader.getInstance(this, mapBoxAccessToken);
+
+                            // Remove the STOP DOWNLOAD ACTION from the notification so that it cannot be pressed and cause the app to crash!
+                            showProgressNotification(mostRecentMapNameUpdate, mostRecentPercentageUpdate, false);
+
+                            // Stop the download first
+                            mapBoxOfflineResourcesDownloader.pauseMapDownload(mapUniqueName, new OnPauseMapDownloadCallback() {
+                                @Override
+                                public void onPauseSuccess() {
+                                    MapBoxOfflineResourcesDownloader
+                                            .getInstance(MapboxOfflineDownloaderService.this, mapBoxAccessToken)
+                                            .deleteMap(mapUniqueName, new OfflineRegion.OfflineRegionDeleteCallback() {
+
+                                                @Override
+                                                public void onDelete() {
+                                                    if (deleteTaskFromRealmDatabase(taskType, mapUniqueName)) {
+                                                        sendBroadcast(SERVICE_ACTION_RESULT.SUCCESSFUL, mapUniqueName, SERVICE_ACTION.STOP_CURRENT_DOWNLOAD);
+
+                                                        showDownloadCompleteNotification("Download for " + mapUniqueName + " stopped!", "All downloaded resources for the map have been deleted also");
+                                                        performNextTask();
```

Comment:
Looks like `performNextTask()` should be executed regardless of the condition. Could you please move it to after the if-else blocks.

Reply:
Yea